client-implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/FieldInfo.java

```diff
@@ -0,0 +1,56 @@
+package io.smallrye.graphql.client.typesafe.impl.reflection;
```

Comment:
Where possible, use Jandex (rather that your own type). Also think about usage in Quarkus, so what can happen during build and what can not. Example, can this be scanned during build ? Then Jandex is perfect for that. (We can change that later)

Reply:
When we add Jandex support, I had a strategy pattern in mind: if it's available, use it; otherwise use reflection. The whole reflection package hides all of this already.

Reply:
If you do it like the server, then the Jandex stuff is in a separate module, and can be included, even in not Quarkus apps. Still easier than using reflection

Reply:
We can change this after merge